.gitlab-ci.yml

```diff
@@ -41,6 +42,7 @@ build-spack-nightlies:
     script:
         # set up spack inside the k4-spack repo
         - git clone --depth 1 https://github.com/spack/spack
+        - if [ -n "$SPACK_VERSION" ]; then cd spack; git checkout $SPACK_VERSION; cd ..; fi 
```

Comment:
Will cloning with `--depth 1` allow to checkout any tag or commit that is not the current `HEAD`?

Reply:
good catch, for this we indeed need a full clone. fixed this now.